### ISSUE TITLE
Failing integration test: throws an error if invalid content-type

### ISSUE THREAD

**Issue by tobiasdiez** [apollo-server-integrations/apollo-server-integration-h3#1]
These are the headers send to `executeHTTPGraphQLRequest`
```ts
{
      'host' => 'localhost:33787',
      'accept-encoding' => 'gzip, deflate',
      'content-type' => 'text/plain',
      'apollo-require-preflight' => 't',
      'content-length' => '36',
      'connection' => 'close'
}
```
This seems to be correct to me, so maybe this is an issue in apollo/server? Or are integrations supposed to filter out non-json requests?

**Comment by trevor-scheer** [comment]
So the error happens in this test:

https://github.com/apollographql/apollo-server/blob/fd48f2ca49ae7992515ec78b62d65f2d9a7921a9/packages/integration-testsuite/src/httpServerTests.ts#L338-L353

And the code that's testing is here:
https://github.com/apollographql/apollo-server/blob/ff69ee5bc2564fa757e5407e76b7bb184b4cd6cb/packages/server/src/runHttpQuery.ts#L128-L132

I'll need to look a bit more closely when I have more time. Can you share the `body` on the `HTTPGraphQLRequest`? I can do some debugging, though probably not until Monday.

**Comment by tobiasdiez** [comment]
Thanks for your help!

I just debugged this. The `body` is the object
```ts
{
  query: "query test{ testString }",
}
```
and thus passes the `isNonEmptyStringRecord` test. I guess the problem was that I always parsed the body as json, regardless of the content-type header. I've fixed that now in https://github.com/apollo-server-integrations/apollo-server-integration-h3/commit/f2d84cd3d4c247b34850f380011d78201b842c1f. Let me know if you think that's not the right way to proceed.
